tests/templates/test_subroutines/test_commuting_evolution.py

```diff
@@ -51,6 +53,39 @@ def evolution_circuit(time):
     assert all(np.isclose(dev1.state, dev2.state))
 
 
+def test_decomposition_expand():
+    """Test that the decomposition of CommutingEvolution is an ApproxTimeEvolution with one step."""
+
+    hamiltonian = 0.5 * qml.PauliX(0) @ qml.PauliY(1)
+    time = 2.345
+
+    op = qml.CommutingEvolution(hamiltonian, time)
+
+    decomp = op.decomposition()
+
+    assert isinstance(decomp, qml.ApproxTimeEvolution)
+    assert all(decomp.hyperparameters["hamiltonian"].coeffs == hamiltonian.coeffs)
+    assert decomp.hyperparameters["n"] == 1
+
+    tape = op.expand()
+    assert len(tape) == 1
+    assert isinstance(tape[0], qml.ApproxTimeEvolution)
+
+
+def test_matrix():

```

Comment:
@antalszava Added this test to check if the actual values are as expected.  Does this get the job done?

I don't want to compare to a hard-coded expected value, as how do I trust the hard-coded expected value?

Reply:
@albi3ro I would be very happy with an additional integration test here, which ensures that the operation gives the correct value in the full QNode pipeline:

```python
@pytest.mark.parametrize("t", [1.0, np.sqrt(2)])
def test_forward_execution(t):
    dev = qml.device('default.qubit', wires=2)

    H = qml.PauliX(0) @ qml.PauliY(1) - 1.0 * qml.PauliY(0) @ qml.PauliX(1)
    freq = (2, 4)

    @qml.qnode(dev)
    def circuit(time):
        qml.PauliX(0)
        qml.CommutingEvolution(H, time, freq )
        return qml.expval(qml.PauliZ(0))

    res = circuit(t)
    expected = -np.cos(t)
    assert np.allclose(res, expected)
```

I know `-np.cos(t)` is hardcoded, but it is analytic and not numeric, and can be easily verified by hand.